From 2ff397498c056b3cafb00092bea20a03dbeb6e59 Mon Sep 17 00:00:00 2001
From: Dinh Nguyen <dinguyen@opensource.altera.com>
Date: Tue, 24 Mar 2015 23:17:28 -0500
Subject: [PATCH 066/142] FogBugz #288412: Add correct reset manager offsets
 for Arria10

There are 2 peripheral module reset manager registers on the Arria10.

Signed-off-by: Dinh Nguyen <dinguyen@opensource.altera.com>
[Original patch taken from
 https://github.com/altera-opensource/linux-socfpga.git
 tag rel_socfpga-4.8_17.03.01_pr]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 arch/arm/mach-socfpga/core.h    |  4 ++++
 arch/arm/mach-socfpga/socfpga.c | 16 ++++++++++++++--
 2 files changed, 18 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-socfpga/core.h b/arch/arm/mach-socfpga/core.h
index 6cf5c9ac2bd2..5f6e1358b6c5 100644
--- a/arch/arm/mach-socfpga/core.h
+++ b/arch/arm/mach-socfpga/core.h
@@ -27,6 +27,10 @@
 
 #define SOCFPGA_A10_RSTMGR_CTRL		0xC
 #define SOCFPGA_A10_RSTMGR_MODMPURST	0x20
+#define SOCFPGA_A10_RSTMGR_CTRL		0xC
+#define SOCFPGA_A10_RSTMGR_PER0MODRST	0x24
+#define SOCFPGA_A10_RSTMGR_PER1MODRST	0x28
+#define SOCFPGA_A10_RSTMGR_BRGMODRST	0x2C
 
 /* System Manager bits */
 #define RSTMGR_CTRL_SWCOLDRSTREQ	0x1	/* Cold Reset */
diff --git a/arch/arm/mach-socfpga/socfpga.c b/arch/arm/mach-socfpga/socfpga.c
index aa4ec13ce1a4..a12be4267517 100644
--- a/arch/arm/mach-socfpga/socfpga.c
+++ b/arch/arm/mach-socfpga/socfpga.c
@@ -35,6 +35,8 @@ void __iomem *sdr_ctl_base_addr;
 unsigned long socfpga_cpu1start_addr;
 void __iomem *clkmgr_base_addr;
 
+static int socfpga_is_a10(void);
+
 static void __init socfpga_soc_device_init(void)
 {
 	struct device_node *root;
@@ -88,8 +90,18 @@ static void __init socfpga_soc_device_init(void)
 
 static void __init enable_periphs(void)
 {
-	/* Release all peripherals from reset.*/
-	__raw_writel(0, rst_manager_base_addr + SOCFPGA_RSTMGR_MODPERRST);
+	if (socfpga_is_a10()) {
+		/* temp hack to enable all periphs from reset for A10 */
+		writel(0x0, rst_manager_base_addr + SOCFPGA_A10_RSTMGR_PER0MODRST);
+		writel(0x0, rst_manager_base_addr + SOCFPGA_A10_RSTMGR_PER1MODRST);
+	} else {
+		writel(0x0, rst_manager_base_addr + SOCFPGA_RSTMGR_MODPERRST);
+	}
+}
+
+static int socfpga_is_a10(void)
+{
+	return of_machine_is_compatible("altr,socfpga-arria10");
 }
 
 void __init socfpga_sysmgr_init(void)
-- 
2.11.0

